src/vmspawn/vmspawn.c

```diff
@@ -0,0 +1,407 @@
+                                return log_oom();
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to set credential from %s: %m", optarg);
+                        arg_settings_mask |= SETTING_CREDENTIALS;
+                        break;
+                }
+
+                case ARG_LOAD_CREDENTIAL: {
+                        r = machine_credential_load(&arg_credentials, &arg_n_credentials, optarg);
+                        if (r == -ENOMEM)
+                                return log_oom();
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to load credential from %s: %m", optarg);
+
+                        arg_settings_mask |= SETTING_CREDENTIALS;
+                        break;
+                }
+
+                case '?':
+                        return -EINVAL;
+
+                default:
+                        assert_not_reached();
+                }
+
+        if (argc > optind) {
+                strv_free(arg_parameters);
+                arg_parameters = strv_copy(argv + optind);
+                if (!arg_parameters)
+                        return log_oom();
+
+                arg_settings_mask |= SETTING_START_MODE;
+        }
+
+        return 1;
+}
+
+static int run_virtual_machine(void) {
+        _cleanup_(ovmf_config_freep) OvmfConfig *ovmf_config = NULL;
+        _cleanup_strv_free_ char **cmdline = NULL;
+        _cleanup_free_ char *machine = NULL, *qemu_binary = NULL, *mem = NULL;
+        int r;
+
+        bool use_kvm = arg_qemu_kvm > 0;
+        if (arg_qemu_kvm < 0) {
+                r = qemu_check_kvm_support();
+                if (r < 0)
+                        return log_error_errno(r, "Failed to check for KVM support: %m");
+                use_kvm = r;
+        }
+
+        r = find_ovmf_config(arg_secure_boot, &ovmf_config);
+        if (r < 0)
+                return log_error_errno(r, "Failed to find OVMF config: %m");
+
+        /* only warn if the user hasn't disabled secureboot */
+        if (!ovmf_config->supports_sb && arg_secure_boot)
+                log_warning("Couldn't find OVMF firmware blob with Secure Boot support, "
+                            "falling back to OVMF firmware blobs without Secure Boot support.");
+
+        const char *accel = use_kvm ? "kvm" : "tcg";
+#ifdef __aarch64__
+        machine = strjoin("type=virt,accel=", accel);
+#else
+        machine = strjoin("type=q35,accel=", accel, ",smm=", on_off(ovmf_config->supports_sb));
+#endif
+        if (!machine)
+                return log_oom();
+
+        r = find_qemu_binary(&qemu_binary);
+        if (r == -EOPNOTSUPP)
+                return log_error_errno(r, "Native architecture is not supported by qemu.");
+        if (r < 0)
+                return log_error_errno(r, "Failed to finq QEMU binary: %m");
+
+        r = asprintf(&mem, "%.4fM", (double)arg_qemu_mem / (1024.0 * 1024.0));
```

Comment:
really? qemu -m takes a fractional value? wowza

Reply:
It does, yes.
Not sure how valuable it is, I can make it just output the closest number of Mb if you think that would be more appropriate?

Reply:
nah, if it reads it sounds fine to me